pkg/driver/node.go

```diff
@@ -163,6 +170,12 @@ func (ns *node) NodeStageVolume(
 
 		logrus.Info("NodeStageVolume: start format and mount operation")
 		if err := ns.formatAndMount(req, devicePath); err != nil {
+			vol.Finalizers = nil
+			// There might still be a case that the attach was successful,
+			// therefore not cleaning up the staging path from CR
+			if _, err = utils.UpdateCStorVolumeAttachmentCR(vol); err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
 			return nil, status.Error(codes.Internal, err.Error())
```

Comment:
err will be nil here  after update call , we can have new error type 
```go
	if _, uerr = utils.UpdateCStorVolumeAttachmentCR(vol); uerr != nil {
		return nil, status.Error(codes.Internal, uerr.Error())
	}
	return nil, status.Error(codes.Internal, err.Error()) 
```

Reply:
Updated